examples/jvm-interop/bridge.c

```diff
@@ -0,0 +1,387 @@
+#include <stdint.h>
+#include <setjmp.h>
+
+#include <jni.h>
+#include "javaSource_Demo.h"
+
+JavaVM* vm;
+
+#define ERR_MSG_MAX_SIZE 256
+
+jmp_buf exception_buffer;
+char* err_msg[ERR_MSG_MAX_SIZE] = {0};
+
+jint JNI_OnLoad(JavaVM *loadedVM, void *reserved)
+{
+    // https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/invocation.html
+    vm = loadedVM;
+    return JNI_VERSION_1_2;
+}
+
+void *roc_alloc(size_t size, unsigned int alignment)
+{
+    return malloc(size);
+}
+
+void *roc_realloc(void *ptr, size_t new_size, size_t old_size,
+                  unsigned int alignment)
+{
+    return realloc(ptr, new_size);
+}
+
+void roc_dealloc(void *ptr, unsigned int alignment)
+{
+    free(ptr);
+}
+
+__attribute__((noreturn)) void roc_panic(void *ptr, unsigned int alignment)
+{
+    if (ptr != NULL) {
+        int ptr_len = strlen((char*)ptr);
+        int len = ptr_len > ERR_MSG_MAX_SIZE ? ERR_MSG_MAX_SIZE : ptr_len;
+        strncpy((char*)err_msg, ptr, (len + 1));
+    }
+    else {
+        strncpy((char *)err_msg, "roc paniced", strlen("roc paniced") + 1);
+    }
+    longjmp(exception_buffer, 1);
+}
+
+
+void *roc_memcpy(void *dest, const void *src, size_t n)
+{
+    return memcpy(dest, src, n);
+}
+
+void *roc_memset(void *str, int c, size_t n)
+{
+    return memset(str, c, n);
+}
+
+// Reference counting
+
+// If the refcount is set to this, that means the allocation is
+// stored in readonly memory in the binary, and we must not
+// attempt to increment or decrement it; if we do, we'll segfault!
+const ssize_t REFCOUNT_READONLY = 0;
+const ssize_t REFCOUNT_ONE = (ssize_t)PTRDIFF_MIN;
+const size_t MASK = (size_t)PTRDIFF_MIN;
+
+// Increment reference count, given a pointer to the first element in a collection.
+// We don't need to check for overflow because in order to overflow a usize worth of refcounts,
+// you'd need to somehow have more pointers in memory than the OS's virtual address space can hold.
+void incref(uint8_t* bytes, uint32_t alignment)
+{
+    ssize_t *refcount_ptr = ((ssize_t *)bytes) - 1;
+    ssize_t refcount = *refcount_ptr;
+
+    if (refcount != REFCOUNT_READONLY) {
+        *refcount_ptr = refcount + 1;
+    }
+}
+
+// Decrement reference count, given a pointer to the first element in a collection.
+// Then call roc_dealloc if nothing is referencing this collection anymore.
+void decref(uint8_t* bytes, uint32_t alignment)
+{
+    if (bytes == NULL) {
+        return;
+    }
+
+    size_t extra_bytes = (sizeof(size_t) >= (size_t)alignment) ? sizeof(size_t) : (size_t)alignment;
+    ssize_t *refcount_ptr = ((ssize_t *)bytes) - 1;
+    ssize_t refcount = *refcount_ptr;
+
+    if (refcount != REFCOUNT_READONLY) {
+        *refcount_ptr = refcount - 1;
+
+        if (refcount == REFCOUNT_ONE) {
+            void *original_allocation = (void *)(refcount_ptr - (extra_bytes - sizeof(size_t)));
+
+            roc_dealloc(original_allocation, alignment);
+        }
+    }
+}
+
+struct RocBytesI32
```

Comment:
nit rename RocBytesI32 to RocListI32 and RocBytesU8 to RocListU8

Reply:
done